src/update.py

```diff
@@ -27,56 +27,99 @@
-(ROOT / "output" / "snapshot").mkdir(parents=True, exist_ok=True)
-(ROOT / "output" / "intermediate").mkdir(parents=True, exist_ok=True)
-(ROOT / "output" / "tables").mkdir(parents=True, exist_ok=True)
-
-if args.profile:
-    profiler = cProfile.Profile()
-    profiler.enable()
-
-# A pipeline chain is any subfolder not starting with "_" in the pipelines folder
-all_pipeline_chains = []
-for item in (ROOT / "src" / "pipelines").iterdir():
-    if not item.name.startswith("_") and not item.is_file():
-        all_pipeline_chains.append(item.name)
-
-# Run all the pipelines and place their outputs into the output folder
-# The output name for each pipeline chain will be the name of the directory that the chain is in
-for pipeline_name in all_pipeline_chains:
-    table_name = pipeline_name.replace("_", "-")
-    if args.only and not table_name in args.only.split(","):
-        continue
-    if args.exclude and table_name in args.exclude.split(","):
-        continue
-    pipeline_chain = DataPipeline.load(pipeline_name)
-    show_progress = not args.no_progress
-    pipeline_output = pipeline_chain.run(
-        pipeline_name, verify=args.verify, process_count=args.process_count, progress=show_progress
+
+def main(
+    output_folder: Path,
+    verify: str = None,
+    only: List[str] = None,
+    exclude: List[str] = None,
+    process_count: int = cpu_count(),
+    show_progress: bool = True,
+) -> None:
+    """
+    Executes the data pipelines and places all outputs into `output_folder`. This is typically
+    followed by publishing of the contents of the output folder to a server.
+
+    Args:
+        output_folder: Root folder where snapshot, intermediate and tables will be placed.
+        verify: Run anomaly detection on the outputs using this strategy. Value must be one of:
+            - None: (default) perform no anomaly detection
+            - "simple": perform only fast anomaly detection
+            - "full": perform exhaustive anomaly detection (can be very slow)
+        only: If provided, only pipelines with a name appearing in this list will be run.
+        exclude: If provided, pipelines with a name appearing in this list will not be run.
+        process_count: Maximum number of processes to use during the data pipeline execution.
+        show_progress: Display progress for the execution of individual DataSources within this
+            pipeline.
+    """
+
+    assert not (
+        only is not None and exclude is not None
+    ), "--only and --exclude options cannot be used simultaneously"
+
+    # Ensure that there is an output folder to put the data in
+    (output_folder / "snapshot").mkdir(parents=True, exist_ok=True)
+    (output_folder / "intermediate").mkdir(parents=True, exist_ok=True)
+    (output_folder / "tables").mkdir(parents=True, exist_ok=True)
+
+    # A pipeline chain is any subfolder not starting with "_" in the pipelines folder
+    all_pipeline_chains = []
```

Comment:
Is `all_pipeline_chains` a list of file names or file paths?

Reply:
`_chains` is a vestige from an old class name, I renamed this variable to be more clear. Done.